fake-vm-component-scripts/fake-vm/launch.py

```diff
@@ -10,25 +10,30 @@
 
 instances = []
 
-def fix_config_keys(source):
-    for k, v in source:
-        yield (k, v) if '.' not in k else (k.split('.')[1], v)
+config_keys = {
+    k: k.split('.')[1] for k in {
+        'configuration.name',
+        'configuration.address',
+        'configuration.login',
+        'configuration.failure',
```

Comment:
It is not a particular aspect of a real vm. I would like to test how name assignment by discovery works, and this is the easiest and the most straightforward way to set names in tests. For example, the following:
```
% /fake-vm/manage.py create --name 'New VM'
```
will result in an instance named 'New VM' in the platform.